crates/worker/src/clients/dynamodb_clients/client.rs

```diff
@@ -0,0 +1,247 @@
+    }
+
+    pub async fn delete_item_attempt(&self, id: &str) -> Result<Option<()>, DBDeleteError> {
+        match_result!(DBDeleteError, self.delete_item(id).await)
+    }
+
+    pub async fn update_item_raw(
+        &self,
+        update_item_builder: &UpdateItemFluentBuilder,
+    ) -> Result<(), DBUpdateError> {
+        let _ = update_item_builder.clone().send().await;
+        Ok(())
+    }
+
+    pub async fn update_item_raw_attempt(
+        &self,
+        update_item_builder: &UpdateItemFluentBuilder,
+    ) -> Result<Option<()>, DBUpdateError> {
+        match_result!(
+            DBUpdateError,
+            self.update_item_raw(update_item_builder).await
+        )
+    }
+
+    pub async fn update_item_status_conditional(
+        &self,
+        id: &str,
+        from: &Status,
+        to: &Status,
+    ) -> Result<(), DBUpdateError> {
+        let _ = self
+            .client
+            .update_item()
+            .table_name(self.table_name.clone())
+            .key(Item::primary_key_name(), AttributeValue::S(id.to_string()))
+            .update_expression("SET #status = :toStatus")
+            .condition_expression("#status = :fromStatus")
+            .expression_attribute_names("#status", Status::attribute_name())
+            .expression_attribute_values(":toStatus", AttributeValue::N(u32::from(to).to_string()))
+            .expression_attribute_values(
+                ":fromStatus",
+                AttributeValue::N(u32::from(from).to_string()),
+            )
+            .send()
+            .await?;
+        Ok(())
+    }
+
+    pub async fn update_item_status_conditional_attempt(
+        &self,
+        id: &str,
+        from: &Status,
+        to: &Status,
+    ) -> Result<Option<()>, DBUpdateError> {
+        match_result!(
+            DBUpdateError,
+            self.update_item_status_conditional(id, from, to).await
+        )
+    }
+
+    pub async fn get_item(&self, key: &str) -> Result<Option<Item>, DBError> {
+        let result = self
+            .client
+            .get_item()
+            .table_name(self.table_name.clone())
+            .key(Item::primary_key_name(), AttributeValue::S(key.to_string()))
+            .send()
+            .await?;
+
+        if let Some(item) = result.item {
+            Ok(Some(item.try_into()?))
+        } else {
+            Ok(None)
+        }
+    }
+
+    pub async fn get_item_attempt(&self, key: &str) -> Result<Option<Option<Item>>, DBError> {
+        match self.get_item(key).await {
+            Err(DBError::GetItemError(err)) => {
+                match_result!(DBGetError, Err(err)).map_err(DBError::from)
+            }
+            result => result.map(|value| Some(value)),
+        }
+    }
+
+    pub async fn scan_items_prior_to(
+        &self,
+        time: &DateTime<Utc>,
+        exclusive_start_key: &Option<HashMap<String, AttributeValue>>,
+    ) -> Result<ScanOutput, DBScanError> {
+        const MAX_CAPACITY: usize = 100;
```

Comment:
```suggestion
        const MAX_CAPACITY: i32 = 100;
```

wdyt?

Reply:
Everything associated with array size, capacity, indexes shall be `usize`